Label "Sizing and text content" documentation and example

arlowhite opened the pull request:
Added a "Sizing and text content" section to the Label API to clarify how text_size and texture_size properties influence text. Link to this section from Button API.
Added an example demonstrating text wrapping: examples/widgets/label_texture_size.py

matham commented on kivy/uix/button.py:
```diff
@@ -12,6 +12,9 @@
 
     button = Button(text='Hello world', font_size=14)
 
+.. note:: To fit the Button to its :attr:`~Button.text`, bind size to :attr:`~kivy.uix.label.Label.texture_size`
```
I don't think this should be here, as we cannot document about parent behavior in every subclass. At some point we have to assume that the user will read the parent doc, rather than duplicate it, or mention it again. Especially, since it's mentioned right above that, that it inherits from Label.

arlowhite replied:
Ok, what do you think about expanding the last sentence instead?

Currently:
> To configure the button, the same properties are used as for the Label class:

Change to:

> To configure the button, the same properties (padding, font_size, etc) and sizing system are used as for the Label class:

matham replied:
That would be fine.

matham commented:
Overall, it's great, however, there are python pep8 style issues. In particular, your lines are too long. Lines are limited to 80 chars. See here how to make git automatically show you the pep8 issues: http://kivy.org/docs/contribute.html#coding-style.

arlowhite commented:
I also realized I haven't discussed how setting size without text_size works.
There is a Note in the text_size attribute doc:
> This text_size property is the same as the usersize property in the Label class. (It is named size= in the constructor.)

But I think it needs to be explained more, so I'll do a bit more work on that.

arlowhite commented:
Ok, I've fixed PEP8 issues, removed the Note from Button and changed the last sentence instead.
I expanded examples/widgets/label_texture_size.py
* size with no text_size, and text_size bound to size examples.
* Shorten, max_lines, valign buttons
* Demonstrates fairly complex layout itself in how the example places buttons

max_lines doesn't actually behave the way I expect, does it have a bug?
Also, the GridLayout doesn't look the same if you shrink the window width and expand it again. Not sure the best way to fix this.

This example is a bit complicated now. Is this an appropriate location and filename for it?

arlowhite commented on kivy/uix/label.py:
```diff
@@ -16,12 +16,50 @@
+    Label:
+        size: self.texture_size
+
+This label's text will be clipped to the specified width and height:
```
I think I want to change this to:
> This label's text will wrap at the specified width and be clipped to the height:

Alternatively, is it better style to recommend this?:
```python
Label:
    size: cm(6), cm(4)
    text_size: self.size
```

arlowhite commented:
I changed that sentence to: "This label's text will wrap at the specified width and be clipped to the height:"
Squashed commits to just two.
I'm done making changes to the branch unless you have any other comments. The only thing I've been debating is whether to rename the example file to something else. Maybe *label_sizing.py*?

arlowhite commented on examples/widgets/label_texture_size.py:
```diff
@@ -0,0 +1,182 @@
+        text: 'Reset'
+        padding: mm(6), mm(3)
+        size: self.texture_size
+        # Setting right this way didn't work, not sure why.
```
This isn't actually a bug, you can't set right and then change width and expect right to stay aligned. I created #3430 and new documentation explains this. I will remove these comments.

arlowhite commented on examples/widgets/label_texture_size.py:
```diff
@@ -0,0 +1,182 @@
+        text: 'Reset'
+        padding: mm(6), mm(3)
+        size: self.texture_size
+        # Align to right side and vertically align-middle with title
```
Removed the comment about binding right not working.
Right-align and vertical-align Reset button using property calculations and link to relevant doc. (Violates PEP8, but I think it's excusable for a URL)

- [ ] The question is do we want to encourage developers to do this? Or should I just change it to a FloatLayout using pos_hint as mentioned in the documentation?

arlowhite commented:
I've decided to split this into two separate new Pull Requests since I have a few questions on the example but the docs are ready to go. Also, this PR is a bit confusing due to the history rewrites.
